keygen/src/keygen.rs

```diff
@@ -349,7 +388,47 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                         .validator(is_parsable::<usize>)
                         .default_value(&default_num_threads)
                         .help("Specify the number of grind threads"),
-                ),
+                )
+                .arg(
+                    Arg::with_name("mnemonic")
+                        .long("mnemonic")
+                        .alias("use-mnemonic")
+                        .help("Generate using a mnemonic key phrase"),
+                )
+                .arg(
+                    Arg::with_name("word_count")
+                        .requires_all(&["mnemonic"])
+                        .long("word-count")
+                        .possible_values(&["12", "15", "18", "21", "24"])
+                        .default_value("12")
+                        .value_name("NUMBER")
+                        .takes_value(true)
+                        .help("(requires mnemonic) Specify the number of words that will be present in the generated seed phrase"),
+                )
+                .arg(
+                    Arg::with_name("language")
+                        .requires_all(&["mnemonic"])
+                        .long("language")
+                        .possible_values(&["english", "chinese-simplified", "chinese-traditional", "japanese", "spanish", "korean", "french", "italian"])
+                        .default_value("english")
+                        .value_name("LANGUAGE")
+                        .takes_value(true)
+                        .help("(requires mnemonic) Specify the mnemonic lanaguage that will be present in the generated seed phrase"),
+                )
```

Comment:
Sure I can do that.  I didn't like the repetition either.  Maybe I should just internally structure 'new' and 'grind' as the same command since they really could just be the same (with the grind variant simply being whether or not you specify a match parameter).  I only retained the separation of these two commands so as not to break backwards compatibility of command names.

If you want to go one step further we could just make 'grind' and 'new' be synonyms, all of their arguments and functionality could duplicate I think without any confusion, and I think they would be backwards compatible too.

Reply:
I think there's just enough different in the processing to justify retaining the two distinct subcommands (mostly around the multi-threaded handling of grind). That said, I do see your point; if it's significantly cleaner with grind and new combined, I'd be happy to take a look.

Reply:
My plan is to do the dedup as you suggested first, and then once that is accepted, follow up with a proposed change to then collapse new and grind to the same command which you can then accept or reject based on your feelings about how it ends up working.  I think this will allow the initial issue to be satisfied via this change, and then my proposal to collapse the commands can be separately evaluated.

One thing I am not certain of - do you want me to pull all of the clap arguments out into the same structure or just the ones that are shared between commands and thus would be duplicated if fully coded in each command?  If I do them all it's more regular but it's unnecessary code for those that are not duplications.  Not sure which you were hoping to see ...

Reply:
>  follow up with a proposed change to then collapse new and grind to the same command 

I don't think we'll want this due to the mutlithreading differences @CriesofCarrots mentioned as well as the potential future improvement of handing off grinding to GPUs.  I think the simplicity of arg parsing logic is worth the distinct commands

Reply:
OK thank you for the feedback.  I will just do the arg de-duping then.

Reply:
> do you want me to pull all of the clap arguments out into the same structure or just the ones that are shared between commands and thus would be duplicated if fully coded in each command?

I think just the ones that are shared, the result being a `mnemonic_args()` method or similar

Reply:
If I could just make one final point just so you know where I'm coming from ...

From my viewpoint, the structure of the command line args is an interface issue, and I think the interface is cleaner and easier to understand if generating keys with prefixes or suffixes was just an additional flag on normal key generation.  From an end-user's perspective they are the same thing: I'm generating a keypair either way, whether or not I require a prefix or suffix is just a refinement on that.

The program internally will of course have to be a bit more complex to map the user's intent to very different implementation techniques depending on the presence or absence of prefix/suffix.  To me that is the natural division of user interface from program internals.  So I would argue that it is better to swallow the extra complexity in implementation for the benefit of more regularity in the command line syntax.  But that's just how I see it.  I am willing to follow the guidelines of the Solana developers here, just wanted to make sure my viewpoint was fully understood.  Thanks for listening!

Reply:
Yep!  I understand and can appreciate your points.  They're not wrong.  There's one more aspect that's communicated by the split of these commands and that's the security of the output.  The keypairs `grind` produces are inherently less secure.  We could stand to improve the messaging here, since now it is only implied by the word "vanity" in the `grind` description.

Reply:
I actually regret adding `grind` into `solana-keygen` (`pubkey` too)
`solana-keygen` ideally is a very small program with minimal external dependencies. 
This ship has mostly sailed though :-/